Documentation/docs.md

```diff
@@ -47,6 +47,7 @@ To learn more about the concepts and internals behind etcd, read the following p
  - [Why etcd][why] (TODO)
  - [Understand data model][data_model]
  - [Understand APIs][understand_apis]
+ - [Understand the design of the Auth subsystem][understand_auth]
```

Comment:
but yea, do you have any suggestions to group this better? my main goal is to make the internal design doc more discoverable. so people who want to learn more can find them.